org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/workspace/GradleClasspathContainerRefresher.java

```diff
@@ -47,22 +46,16 @@ public static void execute(final ExecutionEvent event) {
 
     private static List<IProject> collectSelectedProjects(ExecutionEvent event) {
         ISelection currentSelection = HandlerUtil.getCurrentSelection(event);
-        Builder<IProject> result = ImmutableList.builder();
         if (currentSelection instanceof IStructuredSelection) {
             IStructuredSelection selection = (IStructuredSelection) currentSelection;
-            IAdapterManager adapterManager = Platform.getAdapterManager();
-            for (Object selectionItem : selection.toList()) {
-                @SuppressWarnings({ "cast", "RedundantCast" })
-                IResource resource = (IResource) adapterManager.getAdapter(selectionItem, IResource.class);
-                if (resource != null) {
-                    IProject project = resource.getProject();
-                    if (project != null) {
-                        result.add(project);
-                    }
-                }
-            }
+            @SuppressWarnings("unchecked")
+            ImmutableList<IProject> selectedProjects = FluentIterable.from(selection.toList())
+                    .transform(new AdapterFunction<IProject>(IProject.class, Platform.getAdapterManager()))
+                    .filter(Predicates.hasGradleNature()).toList();
+
+            return selectedProjects;
         }
-        return result.build();
+        return ImmutableList.of();
```

Comment:
The two iterate elements came in by accident and where not intended.
But I am curious what's the benefit of having an else statement here?

On Thu, Aug 27, 2015 at 11:39 PM, Etienne Studer <notifications@github.com>
wrote:

> In
> org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/workspace/GradleClasspathContainerRefresher.java
> <https://github.com/eclipse/buildship/pull/160#discussion_r38149515>:
>
> >          }
> > -        return result.build();
> > +        return ImmutableList.of();
>
> can you put this into an else statement?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/eclipse/buildship/pull/160/files#r38149515>.
>